IsogeomGenerator/generate_isogeom.py

```diff
@@ -0,0 +1,507 @@
+                        'be tagged as float). ' +
+                        'Option can be set more than once to set more tags.'
+                        )
+
+
+def set_shared_options(parser, moab=False):
+    """Set options that are for both the MOAB and VisIt steps.
+
+    Input:
+    ------
+        parser: ArgumentParser object to attach options to
+        moab: bool, True if setting MOAB args (default=False)
+    """
+    set_level_options(parser, moab)
+    parser.add_argument('-db', '--database',
+                        action='store',
+                        nargs=1,
+                        required=False,
+                        default="/tmp",
+                        metavar='DATABASE_PATH',
+                        dest='db',
+                        type=str,
+                        help='Relative path to folder where isosurface ' +
+                        'meshfiles from VisIt are located or to be saved. ' +
+                        'Default is a folder named tmp/ in the current ' +
+                        'directory.'
+                        )
+
+
+def parse_arguments():
+    """Parse user args
+
+    There are three subparsers, one for each mode: full, visit, and moab.
+    Full mode runs both the visit and moab steps. Each parser should have a
+    full help message, simplified usage statement, and examples.
+    """
+    mode_examples = """
+To view full options for each mode, use 'generate_isogeom MODE -h'.
+
+Example usage:
+    (1) Run all the steps start to finish (full mode) starting with meshfile
+        'cw_mesh', scalar data 'wwn', and defining 3 values for the level
+        information at runtime:
+
+        generate_isogeom full cw_mesh wwn -lv 0.1 5.2 12.3
+
+    (2) Run just the first step (visit mode), generating logarithmically spaced
+        levels between 0.1 and 1e+14 and specifying where to write the
+        generated database:
+
+        generate_isogeom visit cw_mesh wwn -gl log -lx 0.1 1e14 -db my_database
+
+    (3) Run only the second step (moab mode), using the levelfile and database
+        from the MOAB step, and specifying a file name for file produced:
+
+        generate_isogeom moab -lf my_database/levelfile -db my_database
+            -g geom1.h5m
+"""
+    mode_description = """
+Use this to generate a full isosurface geometry from a starting Cartesian mesh
+file containing scalar data using VisIt and MOAB. This tool can be run in three
+different modes:
+    full: run both steps starting from the Cartesian mesh file to produce
+        a full DAGMC-compliant isosurface geom. This step first runs the visit
+        step then the moab step.
+    visit: run only the first step using VisIt. This will generate a database
+        of individual mesh isosurfaces from the Cartesian mesh fileself.
+    moab: run only the second step using MOAB. This will generate a full DAGMC-
+        compliant isosurface geometry starting from the database generated from
+        the visit step.
+"""
```

Comment:
why is the formatting different there ?

Reply:
In order to get the line breaks and proper alignment of the string, I have to use `argparse.RawDescriptionHelpFormatter`. This takes whatever is in the docstring as it is formatted to be the text. So if the lines are indented to match the alignment in the code, then they actually appear as indented in the output.